### ISSUE TITLE
lookupMimeType should check for a `null` filename

### ISSUE THREAD

**Issue by donny-dont** [dart-lang/mime#11]
In `http` we just added `mime` as a dependency for support of multipart forms but are hitting an issue with the `lookupMimeType`.

When creating a `MultipartFile` there is an optional `filename` parameter. We also have a `List<int>` present which can be used for detecting magic number. This can result in a call to `lookupMimeType` where `path` is `null`.

Could `lookupMimeType` make both parameters optional and thus add a null check for `path`?

**Comment by gnprice** [comment]
I agree this would be a useful feature. When the application has the bytes of a file but no path, it makes perfect sense to want to use this package to look for a magic-number match.

After all, if there is a match, then `lookupMimeType` (or `MimeTypeResolver.lookup` more generally) would have ignored the path anyway.

When `path` is null, the desired behavior would be that if no magic-number match is found, the lookup should return null, just as it would for a path with no recognized extension.

As a workaround, a caller can write something like `path ?? ''` for the path argument. When no path is available, this passes an empty path. That produces an empty extension, which should never make sense to appear in the extension map; so it should mean the path will effectively be ignored, as desired.
